pandas/core/indexes/base.py

```diff
@@ -1741,10 +1741,9 @@ def append(self, other):
         names = set([obj.name for obj in to_concat])
         name = None if len(names) > 1 else self.name
 
-        if self.is_categorical():
-            # if calling index is category, don't check dtype of others
-            from pandas.core.indexes.category import CategoricalIndex
-            return CategoricalIndex._append_same_dtype(self, to_concat, name)
+        return self._concat(to_concat, name)
+
+    def _concat(self, to_concat, name):
```

Comment:
> instead this only concatenates elements of y

in the end it *is* used to concatenate both y to x, just that this is passed like that in `append` to this helper function. So it is still only used for append.